### PR TITLE
Oxidation log

### LINKED ISSUE #4219: Oxidation log in tests

### PR DESCRIPTION
**Pull request by TimeEngineer**
# What has changed ?

- [ ] Does this PR affect the User ?

closes #4219

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -445,6 +445,8 @@ jobs:
           # By default, `libparsec_crypto` uses Rust Crypto
           cargo test --profile ci-rust --package libparsec_crypto --features use-sodiumoxide
         timeout-minutes: 30
+        env:
+          RUST_LOG: trace
```
Comment (FirelightFlagboy) [inline .github/workflows/ci.yml:449]: Should we use the trace level ? Won't it produce too much noise ?

**Review by FirelightFlagboy — COMMENTED** [review]
Can you split this PR into 2 ?

- One that only add the log support
- One that add the log support in test

**Comment by TimeEngineer** [comment]
The problem is that if I split, we would not have the println in the tests, so log are not used

File: .github/workflows/ci.yml (hunk 1 above)
Comment (TimeEngineer) [reply .github/workflows/ci.yml]: I set debug then

**Review by FirelightFlagboy — APPROVED** [review]

**Comment by FirelightFlagboy** [comment]
@TimeEngineer Could you try if we have the log output when a rust test timeout ?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -441,10 +441,12 @@ jobs:
         shell: bash
         run: |
           set -ex
-          cargo test --profile ci-rust --workspace ${{ env.cargo-exclude-unused-crates }}
+          cargo test --profile ci-rust --workspace ${{ env.cargo-exclude-unused-crates }} -- --nocapture
```
Comment (FirelightFlagboy) [inline .github/workflows/ci.yml:444]: There is a way to only show the output log for failing tests only (included the tests that time out) ?

**Review by FirelightFlagboy — CHANGES_REQUESTED** [review]
I've looked at the tests output `--no-capture` isn't something we want when running all tests

File: .github/workflows/ci.yml (hunk 2 above)
Comment (TimeEngineer) [reply .github/workflows/ci.yml:444]: I don't find

Comment (FirelightFlagboy) [reply .github/workflows/ci.yml:444]: The question can be rewritten as:

What append when a test fail without using `--nocapture` ?

Comment (TimeEngineer) [reply .github/workflows/ci.yml]: It logs nothing in case of timeout

File: oxidation/libparsec/crates/platform_local_db/src/executor.rs [hunk 3]
```diff
@@ -200,17 +199,22 @@ impl BackgroundSqliteExecutor {
...
     use crate::DatabaseError;
     use diesel::{connection::SimpleConnection, Connection, SqliteConnection};
+    use rstest::rstest;
+    use std::time::Duration;
 
     use super::SqliteExecutor;
 
-    #[tokio::test]
+    #[rstest]
+    #[timeout(Duration::new(60, 0))]
```
Comment (FirelightFlagboy) [inline oxidation/libparsec/crates/platform_local_db/src/executor.rs:216]:
```suggestion
    #[timeout(Duration::new(30, 0))]
```

`30sec` to `open -> close -> re-open` a file should be plenty enough

**Review by FirelightFlagboy — CHANGES_REQUESTED** [review]

**Review by touilleMan — APPROVED** [review]
Should investigate a way to avoid using test-log decorator for each function (typically configuring once the log in the main of the tests)

File: oxidation/libparsec/crates/platform_local_db/src/executor.rs (hunk 3 above)
Comment (TimeEngineer) [reply oxidation/libparsec/crates/platform_local_db/src/executor.rs]: Done in another PR

**Pull request merged**